js/src/logger.ts

```diff
@@ -3215,6 +3279,36 @@ export class SpanImpl implements Span {
     let sanitizedAndInternalData = { ...internalData };
     mergeDicts(sanitizedAndInternalData, sanitized);
 
+    const serializableInternalData: typeof sanitizedAndInternalData = {};
+    const lazyInternalData: Record<string, LazyValue<unknown>> = {};
+
+    for (const [key, value] of Object.entries(sanitizedAndInternalData) as [
```

Comment:
Do we care about pulling out streams in nested components of the object, such as within the `metadata`? I wonder if it would be ideal to stick this logic into the `serializedPartialRecord` json `replacer` function, where if we encounter a stream, we replace it with `undefined` and store the stream in `lazyInternalData`. And then deep-merge it back in?

Reply:
Maybe but also would prefer not to over engineer it until we know more about the pattern.